Please review the entire Question.java file with special attention to…

esengoba opened the pull request:
Look at these methods: setCategoryPathName() and selectQuestion()

esengoba committed:
Please review the entire Question.java file with special attention to the methods:
setCategoryPathName() and selectQuestion()

fmjosuel commented on src/model/Question.java:
```diff
@@ -141,35 +141,29 @@ public void setCategoryPathName(int path) {
      public int selectQuestion(){
          Random random = new Random();
 
          if (categoryID == 1) { //Geek Out
```
Isn't a switch statement suitable here? Instead of if/else's?

esengoba replied:
Addressed in new pull request.

fmjosuel commented on src/model/Question.java:
```diff
@@ -141,35 +141,29 @@ public void setCategoryPathName(int path) {
          if (categoryID == 1) { //Geek Out
-             range = (int )(Math.random() * (120 - 101) + 101);
-             System.out.println("hi1 " +range);
+             randomID = (int )(Math.random() * (120 - 101) + 101);
```
`randomID = rand.nextInt(120) + 101;`
can alternatively do the job here. Kinda looks good, in my eyes!

esengoba replied:
// We'll address this in a later pull request -- will require refactoring later to capture the number of questions in a given file

fmjosuel commented on src/model/Question.java:
```diff
@@ -141,35 +141,29 @@ public void setCategoryPathName(int path) {
+             randomID = (int )(Math.random() * (605 - 601) + 601);
          }
-         return range;
+         return randomID;
```
Also, can't we make these numbers dynamic? Say for example we decide to add questions in each category, we'd need to come back and change each number here, which is costly, in my opinion.

fmjosuel replied:
But otherwise this does the intended job!

fmjosuel reviewed (commented):
Just a few comments!

esengoba committed:
Please review the entire Question.java file with special attention to the methods:
setCategoryPathName() and selectQuestion()

Updated the pathnames to switch statements
